packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx

```diff
@@ -0,0 +1,731 @@
+import React, { useEffect } from 'react'; // eslint-disable-line
+import './registration_form.css';
+import { useForm } from 'react-hook-form';
+import { useState } from 'react';
+import axios from 'axios';
+import { url } from '../../../../Global';
+
+const RegistrationForm = () => {
+    const {
+        register,
+        handleSubmit,
+        setError,
+        formState: { errors } // eslint-disable-line
+    } = useForm({ mode: 'all' });
+
+    const [loadingAnimation, setLoadingAnimation] = useState(false);
+    const [submitPressed, setSubmitPressed] = useState(false);
+    const [submitButtonValue, setSubmitButtonValue] = useState('register'); // eslint-disable-line
+    const [apiError, setApiError] = useState(false);
+
+    const makeRegistration = (data) => {
+        axios({
+            method: 'post',
+            url: url + '/api/hackathon/register',
+            headers: {
+                'BEARER-TOKEN': localStorage.getItem('auth_token')
+            },
+            data: { ...data }
+        })
+            // eslint-disable-next-line no-unused-vars
+            .then((res) => {
+                console.log(res);
+                // console.log('New registartion was added');
+            })
+            .catch((err) => {
+                console.log(err);
+                alert(err['response']['data']['data']['data']);
+            });
+    };
+    const convertToBool = (data) => {
+        data.prevhackaubgparticipation =
+            data.prevhackaubgparticipation == 'False' ? false : true;
```

Comment:
instead of doing this, come up with a function which does something similar: 

```javascript

parseVars(data){
    Object.entries(data).forEach(field => {
      if (field[1] == "True" || field[1] == "False") {
             data[field[0]] = field[1] == "True"? true : false
        }
    }

  return data
}
```

then call this function in `onSubmit`:

```javascript

const onSubmit = (data) => {
         setLoadingAnimation(true);
         data = parseVars(data);
         makeRegistration(data);
```

be advised: this is from the top of my hand, there is the possibility something is wrong in this code